packages/wonder-blocks-modal/src/components/flexible-panel.tsx

```diff
@@ -0,0 +1,211 @@
+     * to the `ModalLauncher`.  Doing so will throw an error.
+     */
+    onClose?: () => unknown;
+    /**
+     * Test ID used for e2e testing.
+     *
+     * In this case, this `testId` comes from the `testId` prop defined in the
+     * Dialog variant (e.g. OnePaneDialog).
+     */
+    testId?: string;
+};
+
+/**
+ * FlexiblePanel is the content container.
+ *
+ * **Implementation notes:**
+ *
+ * If you are creating a custom Dialog, make sure to follow these guidelines:
+ * - Make sure to add this component inside the [FlexibleDialog](/#flexibledialog).
+ * - If you need to create e2e tests, make sure to pass a `testId` prop. This
+ *   will be passed down to this component using a sufix: e.g.
+ *   `some-random-id-FlexiblePanel`. This scope will be propagated to the
+ *   CloseButton element as well: e.g. `some-random-id-CloseButton`.
+ *
+ * ```js
+ * <FlexibleDialog>
+ *      <FlexiblePanel content={"custom content goes here"} />
+ * </FlexibleDialog>
+ * ```
+ */
+export default function FlexiblePanel({
+    backgroundStyles,
+    closeButtonVisible = true,
+    scrollOverflow = true,
+    content,
+    titleId,
+    title,
+    onClose,
+    style,
+    testId,
+}: Props) {
+    const renderMainContent = React.useCallback((): React.ReactNode => {
+        const contentNode =
+            typeof content === "function" ? (
+                content({title})
+            ) : (
+                <>
+                    {title}
+                    {content}
+                </>
+            );
+        const mainContent = ModalContent.isComponentOf(contentNode) ? (
+            (contentNode as React.ReactElement<PropsFor<typeof ModalContent>>)
+        ) : (
+            <ModalContent>{contentNode}</ModalContent>
+        );
+
+        if (!mainContent) {
+            return mainContent;
+        }
+
+        return React.cloneElement(mainContent, {
+            // Pass the scrollOverflow and header in to the main content
+            scrollOverflow,
+            // We override the styling of the main content to help position
+            // it if there is a close button being
+            // shown. We have to do this here as the ModalContent doesn't
+            // know about things being positioned around it.
+            style: [mainContent.props.style],
+        });
+    }, [title, content, scrollOverflow]);
+
+    const mainContent = renderMainContent();
+
+    const defaultBackgroundStyle = {
+        backgroundColor: semanticColor.surface.primary,
+    };
+
+    const combinedBackgroundStyles = {
+        ...defaultBackgroundStyle,
+        ...backgroundStyles,
+    };
+
+    return (
+        <View
+            style={[styles.wrapper, combinedBackgroundStyles, style]}
+            testId={testId && `${testId}-panel`}
+        >
+            {closeButtonVisible && (
+                <CloseButton
+                    onClick={onClose}
+                    style={[styles.closeButton]}
```

Comment:
Should we support custom styling for the close button? There could be a case where we want the close icon to have an inverse color if a full width background image is darker! If so, it would make sense to have a `styles` prop instead of `style` so that different elements can be styled!

Reply:
Added!